### PR TITLE
Adding list of pods

### PR DESCRIPTION
**Pull request by adfost**

### CODE REVIEW

**Comment by dvaldivia** [comment]
If I run other applications on the same namespace as the Tenant they get listed, please restrict the pods to only Tenant Pods

Screen Shot 2021-05-04 at 9 32 01 AM

**Review by dvaldivia — CHANGES_REQUESTED** [review]
See screenshot

File: portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx [hunk 1]
```diff
@@ -653,6 +682,37 @@ const TenantDetails = ({
               </Grid>
             </React.Fragment>
           )}
+          {selectedTab === 2 && pods[0].time && (
+            <TableWrapper
+              itemActions={[
+                {
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx:688]: The delete action is not yet implemented, please remove this option

**Comment by dvaldivia** [comment]
Can you also attach a screenshot of the final screen to this ticket

File: portal-ui/src/screens/Console/Tenants/ListTenants/types.ts [hunk 2]
```diff
@@ -28,6 +28,16 @@ export interface IPool {
   label?: string;
 }
 
+export interface IPod {
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/ListTenants/types.ts:31]: We should call it `IPodListElement` to have more clarity on what does this interface do

File: portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx [hunk 3]
```diff
@@ -297,6 +302,28 @@ const TenantDetails = ({
...
+        `/api/v1/namespaces/${tenantNamespace}/tenants/${tenantName}/pods`
+      )
+      .then((result: IPod[]) => {
+        for (let i = 0; i < result.length; i++) {
+          let currentTime = new Date().getSeconds();
+          console.log(currentTime);
+          result[i].time = niceDays(
+            (currentTime - parseInt(result[i].timecreated)).toString()
+          );
+          console.log(result[i]);
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx:318]: Please remove console.log

File: portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx (hunk 1 above)
Comment (bexsoft) [reply portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx:688]: You can send an empty array to item actions, it is not necessary to have an option here always

Comment (bexsoft) [reply portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx:688]: You can send an empty array to item actions, it is not necessary to have an option here always

File: swagger.yml [hunk 4]
```diff
@@ -3835,6 +3862,29 @@ definitions:
...
+    required:
+      - name
+    properties:
+      name:
+        type:
+          string
+      status:
+        type:
+          string
+      timecreated:
```
Comment (bexsoft) [inline swagger.yml:3877]: Please use camel case:
```suggestion
      timeCreated:
```

File: swagger.yml [hunk 5]
```diff
@@ -3835,6 +3862,29 @@ definitions:
...
+        type: integer
+      podip:
```
Comment (bexsoft) [inline swagger.yml:3879]: Same camel case here:
```suggestion
      podIP:
```

File: portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx [hunk 6]
```diff
@@ -297,6 +302,28 @@ const TenantDetails = ({
...
+  const loadPods = () => {
+    api
+      .invoke(
+        "GET",
+        `/api/v1/namespaces/${tenantNamespace}/tenants/${tenantName}/pods`
+      )
+      .then((result: IPod[]) => {
+        for (let i = 0; i < result.length; i++) {
+          let currentTime = new Date().getSeconds();
+          console.log(currentTime);
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx:314]: Please remove console.log

File: portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx [hunk 7]
```diff
@@ -297,6 +302,28 @@ const TenantDetails = ({
...
+          result[i].time = niceDays(
+            (currentTime - parseInt(result[i].timecreated)).toString()
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx:316]: Not sure if this what is intended, as you are receiving an epoch time from backend and you are substracting the amount of seconds of current time:

Screen Shot 2021-05-04 at 12 37 21

**Review by bexsoft — CHANGES_REQUESTED** [review]

File: restapi/admin_tenants.go [hunk 8]
```diff
@@ -1336,6 +1344,32 @@ func getTenantUsageResponse(session *models.Principal, params admin_api.GetTenan
...
+		return nil, prepareError(err)
+	}
+	listOpts := metav1.ListOptions{}
+	pods, err := clientset.CoreV1().Pods(params.Namespace).List(ctx, listOpts)
+	if err != nil {
+		return nil, prepareError(err)
+	}
+	retval := []*models.TenantPod{}
+	for i := range pods.Items {
+		if pods.Items[i].Name[0:len(params.Tenant)] == params.Tenant {
```
Comment (dvaldivia) [inline restapi/admin_tenants.go:1360]: I broke this logic by deploying a pod for wodpress with the tenant prefix

File: admin.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (dvaldivia) [inline admin.json:1]: remove this file

File: restapi/admin_tenants.go [hunk 10]
```diff
@@ -1336,6 +1344,43 @@ func getTenantUsageResponse(session *models.Principal, params admin_api.GetTenan
...
 }
 
+func getTenantPodsResponse(session *models.Principal, params admin_api.GetTenantPodsParams) ([]*models.TenantPod, *models.Error) {
+	ctx := context.Background()
+	clientset, err := cluster.K8sClient(session.STSSessionToken)
+	if err != nil {
+		return nil, prepareError(err)
+	}
+	listOpts := metav1.ListOptions{
+		TypeMeta:             metav1.TypeMeta{},
```
Comment (dvaldivia) [inline restapi/admin_tenants.go:1354]: remove all unused fields, also this doesn't filter anything like we discussed

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Review by dvaldivia — DISMISSED** [review]

File: portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx [hunk 11]
```diff
@@ -297,6 +302,26 @@ const TenantDetails = ({
...
+        for (let i = 0; i < result.length; i++) {
+          let currentTime = new Date().getSeconds();
+          result[i].time = niceDays(
+            (currentTime - parseInt(result[i].timeCreated)).toString()
+          );
+        }
+        setPods(result);
+      })
+      .catch((err) => {
+        console.log("Error in loading pods");
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/TenantDetails.tsx:321]: Should we show the error alert here?

```suggestion
        setErrorSnackMessage("Error loading pods")
```

**Review by bexsoft — CHANGES_REQUESTED** [review]

**Review by bexsoft — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**